javascripts/discourse/lib/sidebar-panel.js

```diff
@@ -0,0 +1,145 @@
+import { cached } from "@glimmer/tracking";
+import { computed } from "@ember/object";
+import BaseCustomSidebarSection from "discourse/lib/sidebar/base-custom-sidebar-section";
+import BaseCustomSidebarSectionLink from "discourse/lib/sidebar/base-custom-sidebar-section-link";
+import DiscourseURL from "discourse/lib/url";
+import { getOwnerWithFallback } from "discourse-common/lib/get-owner";
+import { getAbsoluteURL, samePrefix } from "discourse-common/lib/get-url";
+import I18n from "discourse-i18n";
+import { SIDEBAR_DOCS_PANEL } from "../services/docs-sidebar";
+import { normalizeName } from "./utils";
+
+const sidebarPanelClassBuilder = (BaseCustomSidebarPanel) =>
+  class DocsSidebarPanel extends BaseCustomSidebarPanel {
+    key = SIDEBAR_DOCS_PANEL;
+    hidden = true;
+
+    get docsSidebar() {
+      return getOwnerWithFallback(this).lookup("service:docs-sidebar");
+    }
+
+    @cached
+    get sections() {
+      const router = getOwnerWithFallback(this).lookup("service:router");
+
+      return this.docsSidebar.sectionsConfig.map((config) => {
+        return prepareDocsSection({ config, router });
+      });
+    }
+
+    get filterable() {
+      return !this.docsSidebar.loading;
+    }
+  };
+
+export default sidebarPanelClassBuilder;
+
+function prepareDocsSection({ config, router }) {
+  return class extends BaseCustomSidebarSection {
+    #config = config;
+
+    get sectionLinks() {
+      return this.#config.links;
+    }
+
+    get name() {
+      return this.#config.name;
+    }
+
+    get title() {
+      return this.#config.text;
+    }
+
+    get text() {
+      return this.#config.label
+        ? I18n.t(this.#config.label)
+        : this.#config.text;
+    }
+
+    get links() {
+      return this.sectionLinks.map(
+        (sectionLinkData) =>
+          new SidebarDocsSectionLink({
+            data: sectionLinkData,
+            router,
+          })
+      );
+    }
+
+    get displaySection() {
+      return true;
+    }
+
+    get hideSectionHeader() {
+      return this.#config.name === "root";
+    }
+
+    get collapsedByDefault() {
+      return false;
+    }
+  };
+}
+
+class SidebarDocsSectionLink extends BaseCustomSidebarSectionLink {
+  #data;
+
+  constructor({ data, router }) {
+    super(...arguments);
+
+    this.#data = data;
+    this._router = router;
+  }
+
+  @computed("_router.currentRoute", "href")
+  get active() {
+    if (DiscourseURL.isInternal(this.href) && samePrefix(this.href)) {
+      const topicRouteInfo = this._router
+        .recognize(this.href.replace(getAbsoluteURL("/"), "/"), "")
+        .find((v) => v.name === "topic");
+
+      const currentTopicRouteInfo = this._router.currentRoute.find(
```

Comment:
```suggestion
        .findBy("name", "topic");
```

Reply:
This is not an array. `.findBy` doesn't work here. Ember returns a `RouteInfo` instance  that contains a `.find(...)` method.

Reply:
Ah my bad, no probs